core/src/main/java/io/questdb/log/LogRecordUtf8Sink.java

```diff
@@ -197,8 +197,13 @@ public int size() {
     }
 
     @Override
-    public void toSink(@NotNull CharSinkBase<?> sink) {
-        Utf8s.utf8ToUtf16(address, _wptr, sink);
+    public void toSink(@NotNull CharSink<?> sink) {
+        if (sink instanceof Utf16Sink) {
```

Comment:
nit: it might be a better idea to expose a `getStorageFormat()` method on `CharSink` which would return UTF-8 or UTF-16 int code. There is a penalty introduced by `instanceof` checks, so we should avoid them when possible.

Reply:
fixed in https://github.com/questdb/questdb/pull/4137/commits/645c36d9aeea1a38fc8b3a9f131853e4c5b0b9aa